src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -28,14 +43,31 @@ def _resolve_strict_dependencies(target):
 
 def _resolve_exports(target):
   for export in getattr(target, 'exports', []):
+    if not isinstance(export, Target):
+      addr = Address.parse(export, relative_to=target.address.spec_path)
+      export = target._build_graph.get_target(addr)
+      if export is None:
```

Comment:
I think `export is None` and `export not in target.dependencies` should both be treated as the same error: if it isn't in the graph, it definitely wasn't in your dependencies list.